src/metabase/lib/field.cljc

```diff
@@ -132,11 +132,11 @@
                   (when-let [unit (:temporal-unit opts)]
                     {::temporal-unit unit})
                   (cond
-                    (integer? id-or-name) (resolve-field-id query stage-number id-or-name)
+                    (integer? id-or-name) (or (resolve-field-id query stage-number id-or-name)
+                                              {:lib/type :metadata/column, :name id-or-name})
                     join-alias            {:lib/type :metadata/column, :name id-or-name}
```

Comment:
This is a very conservative change. I don't know how (much) it would hurt to try and resolve fields by name even if they have `join-alias` set.